Failures for new benefit-repeal parameters in test_compatible_data.py

The need to resolve issue #1846 has been highlighted by the merge of pull request #1719 that adds several benefit-repeal policy parameters.  The lack of compatible-data test documentation means that very few developers know how to make the tests work for new policy parameters.

@MattHJensen, could you fix these test failures and add documentation as you do that?
Otherwise, you'll have to do this every time a new policy parameter is added. 
Thanks very much.

Here are the messages from the four test failures:
```
_________________________ test_compatible_data[True-4] _________________________
        if errors:
            print(errors)
>           assert 'compatible_data' == 'invalid'
E           AssertionError: assert 'compatible_data' == 'invalid'
E             - compatible_data
E             + invalid
tests/test_compatible_data.py:257: AssertionError
----------------------------- Captured stdout call -----------------------------
You loaded data for 2009.
Tax-Calculator startup automatically extrapolated your data to 2013.
You loaded data for 2009.
Tax-Calculator startup automatically extrapolated your data to 2013.
You loaded data for 2009.
Tax-Calculator startup automatically extrapolated your data to 2013.
You loaded data for 2009.
Tax-Calculator startup automatically extrapolated your data to 2013.
You loaded data for 2009.
Tax-Calculator startup automatically extrapolated your data to 2013.
ERROR: _BEN_Unemployment_repeal not True for puf
ERROR: _BEN_OASDI_repeal not True for puf

________________________ test_compatible_data[False-4] _________________________
        if errors:
            print(errors)
>           assert 'compatible_data' == 'invalid'
E           AssertionError: assert 'compatible_data' == 'invalid'
E             - compatible_data
E             + invalid
tests/test_compatible_data.py:257: AssertionError
----------------------------- Captured stdout call -----------------------------
You loaded data for 2014.
Tax-Calculator startup automatically extrapolated your data to 2014.
You loaded data for 2014.
Tax-Calculator startup automatically extrapolated your data to 2014.
You loaded data for 2014.
Tax-Calculator startup automatically extrapolated your data to 2014.
You loaded data for 2014.
Tax-Calculator startup automatically extrapolated your data to 2014.
You loaded data for 2014.
Tax-Calculator startup automatically extrapolated your data to 2014.
ERROR: _BEN_Other_repeal not False for cps
ERROR: _BEN_MCARE_repeal not False for cps
ERROR: _BEN_SNAP_repeal not False for cps
ERROR: _BEN_SSI_repeal not False for cps
ERROR: _BEN_MCAID_repeal not False for cps

________________________ test_compatible_data[False-5] _________________________
        if errors:
            print(errors)
>           assert 'compatible_data' == 'invalid'
E           AssertionError: assert 'compatible_data' == 'invalid'
E             - compatible_data
E             + invalid
tests/test_compatible_data.py:257: AssertionError
----------------------------- Captured stdout call -----------------------------
You loaded data for 2014.
Tax-Calculator startup automatically extrapolated your data to 2014.
You loaded data for 2014.
Tax-Calculator startup automatically extrapolated your data to 2014.
You loaded data for 2014.
Tax-Calculator startup automatically extrapolated your data to 2014.
You loaded data for 2014.
Tax-Calculator startup automatically extrapolated your data to 2014.
ERROR: _BEN_Vet_repeal not False for cps

________________________ test_compatible_data[False-11] ________________________
        if errors:
            print(errors)
>           assert 'compatible_data' == 'invalid'
E           AssertionError: assert 'compatible_data' == 'invalid'
E             - compatible_data
E             + invalid
tests/test_compatible_data.py:257: AssertionError
----------------------------- Captured stdout call -----------------------------
You loaded data for 2014.
Tax-Calculator startup automatically extrapolated your data to 2014.
You loaded data for 2014.
Tax-Calculator startup automatically extrapolated your data to 2014.
You loaded data for 2014.
Tax-Calculator startup automatically extrapolated your data to 2014.
You loaded data for 2014.
Tax-Calculator startup automatically extrapolated your data to 2014.
You loaded data for 2014.
Tax-Calculator startup automatically extrapolated your data to 2014.
You loaded data for 2014.
Tax-Calculator startup automatically extrapolated your data to 2014.
ERROR: _ID_BenefitCap_Switch not False for cps
```

--- a/taxcalc/tests/test_compatible_data.py
+++ b/taxcalc/tests/test_compatible_data.py
@@ -72,7 +72,17 @@ TEST_YEAR = 2020
 @pytest.fixture(scope='module', name='reform_xx')
 def fixture_reform_xx():
     """
-    Fixture for reform dictionary where reform starts before TEST_YEAR
+    Fixture for reform dictionary where reform starts before TEST_YEAR.
+
+    The provisions in the baseline reform, designated in _reform_xx,
+    is designed to activate parameters that are inactive under current law.
+    For example a phaseout rate for a new credit is inactive is the credit's
+    amount is set to zero under current law. In order to activate the phaseout
+    rate, the credit amount should be set above zero. The provisions interact
+    with each other: you may acidentally deactivate one parameter
+    by introducing a provision to activate another. If you find that a pair of
+    parameters are impossible test jointly, add one to the local variable
+    `exempt_from_testing` in `test_compatible_data()` as a last resort.
     """
     assert XX_YEAR < TEST_YEAR
 

--- a/taxcalc/tests/test_compatible_data.py
+++ b/taxcalc/tests/test_compatible_data.py
@@ -75,7 +75,7 @@ def fixture_reform_xx():
     Fixture for reform dictionary where reform starts before TEST_YEAR.
 
     The provisions in the baseline reform, designated in _reform_xx,
-    is designed to activate parameters that are inactive under current law.
+    are chosen to activate parameters that are inactive under current law.
     For example a phaseout rate for a new credit is inactive is the credit's
     amount is set to zero under current law. In order to activate the phaseout
     rate, the credit amount should be set above zero. The provisions interact

--- a/taxcalc/tests/test_compatible_data.py
+++ b/taxcalc/tests/test_compatible_data.py
@@ -76,7 +76,7 @@ def fixture_reform_xx():
 
     The provisions in the baseline reform, designated in _reform_xx,
     are chosen to activate parameters that are inactive under current law.
-    For example a phaseout rate for a new credit is inactive is the credit's
+    For example a phaseout rate for a new credit is inactive if the credit's
     amount is set to zero under current law. In order to activate the phaseout
     rate, the credit amount should be set above zero. The provisions interact
     with each other: you may acidentally deactivate one parameter

--- a/taxcalc/tests/test_compatible_data.py
+++ b/taxcalc/tests/test_compatible_data.py
@@ -112,7 +112,7 @@ def fixture_reform_xx():
             '_AGI_surtax_trt': [0.5],
             '_ID_AmountCap_rt': [0.9],
             '_II_brk7': [[1000000, 1000000, 1000000, 1000000, 1000000]],
-            '_ID_BenefitCap_rt': [0.4],
+            '_ID_BenefitCap_rt': [0.3],
             '_PT_rt7': [.35],
             '_II_em': [1000],
             '_ID_Casualty_hc': [.1],

